src/python/pants/engine/target.py

```diff
@@ -1216,23 +1220,9 @@ def compute_value(
         return FrozenDict(result)
 
 
-# -----------------------------------------------------------------------------------------------
-# Sources and codegen
-# -----------------------------------------------------------------------------------------------
-
-
-class Sources(AsyncField):
-    """A list of files and globs that belong to this target.
-
-    Paths are relative to the BUILD file's directory. You can ignore files/globs by prefixing them
-    with `!`. Example: `sources=['example.py', 'test_*.py', '!test_ignore.py']`.
-    """
-
-    alias = "sources"
+class AsyncStringSequenceField(AsyncField):
```

Comment:
No, but it avoids duplicating the exact same code 3 times. We arguably should have factored this up earlier, but I was trying to avoid premature abstraction originally. Now that we have 3 async fields using this same pattern, it seemed cleanest to factor it up.

Note that we're not yet adding `AsyncBoolField`, `AsyncStringField`, etc. Async fields are very rare, and it would be premature to add those imo. But `AsyncStringSequenceField` is used a lot.